Std/Data/Nat/Lemmas.lean

```diff
@@ -133,6 +131,26 @@ theorem recDiagOn_succ_succ {motive : Nat → Nat → Sort _} (zero_zero : motiv
     (succ_succ : ∀ m n, motive (m+1) (n+1)) (m n) :
     Nat.casesDiagOn (m+1) (n+1) zero_zero zero_succ succ_zero succ_succ = succ_succ m n := rfl
 
+/-! ### Associative and commutative instance. -/
+
+instance : Std.IsAssociative (α := Nat) (· + ·) where
+  assoc := Nat.add_assoc
+
+instance : Std.IsCommutative (α := Nat) (· + ·) where
+  comm := Nat.add_comm
+
+instance : Std.IsCommId (α := Nat) (· + ·) 0 where
+  right_id := Nat.add_zero
+
+instance : Std.IsAssociative (α := Nat) (· * ·) where
+  assoc := Nat.mul_assoc
+
+instance : Std.IsCommutative (α := Nat) (· * ·) where
+  comm := Nat.mul_comm
+
+instance : Std.IsCommId (α := Nat) (· * ·) 1 where
+  right_id := Nat.mul_one
+
```

Comment:
These could be in Basic? I'm sincerely asking, this is debatable.

Reply:
I would be inclined not to unless it appears in a definition. I don't think it does in this case, because list folds don't require any properties on the operation (unlike finset folds)

Reply:
`List.fold` does have a typeclass dependency on `HasId` in this implementation to find the identity (there aren't any properties though in that class -- it's essentially a way to associate an identity value to a function.

I think that's a slight argument for putting them in `Basic` so that the typeclass instance is available (via `IsCommId`).  I kept it here though as they are almost entirely for verification though.